lib/checkunusedvar.cpp

```diff
@@ -1586,3 +1589,61 @@ bool CheckUnusedVar::isEmptyType(const Type* type)
     emptyType=false;   // unknown types are assumed to be nonempty
     return emptyType;
 }
+
+bool CheckUnusedVar::isFunctionWithoutSideEffects(const Function& func)
+{
+    // no body to analyze
+    if (!func.hasBody() || !func.functionScope || !func.functionScope->bodyStart) {
+        return false;
+    }
+
+    Token* bodyToken = func.functionScope->bodyStart->next();
+    while (bodyToken != func.functionScope->bodyEnd) {
```

Comment:
I prefer a for loop here, that will guarantee bodyToken is moved each loop even if there would be a continue somewhere or something..